jetty-websocket/javax-websocket-server/src/main/java/org/eclipse/jetty/websocket/javax/server/JavaxWebSocketServerContainer.java

```diff
@@ -71,34 +80,86 @@ public static WebSocketContainer getWebSocketContainer()
+
+            Executor executor = httpClient == null?null:httpClient.getExecutor();
+            if (executor == null)
+                executor = (Executor)servletContext
+                    .getAttribute("org.eclipse.jetty.server.Executor");
+            if (executor == null)
+                executor = contextHandler.getServer().getThreadPool();
+
+            if (httpClient != null && httpClient.getExecutor() == null)
+                httpClient.setExecutor(executor);
+
+            // Create the Jetty ServerContainer implementation
+            container = new JavaxWebSocketServerContainer(
+                WebSocketMapping.ensureMapping(servletContext), httpClient, executor);
+            contextHandler.addManaged(container);
+            contextHandler.addLifeCycleListener(container);
+        }
+        // Store a reference to the ServerContainer per - javax.websocket spec 1.0 final - section 6.4: Programmatic Server Deployment
+        servletContext.setAttribute(ServerContainer.class.getName(), container);
+        return container;
     }
 
-    private final WebSocketCreatorMapping _webSocketCreatorMapping;
+    private final WebSocketMapping _webSocketMapping;
     private final JavaxWebSocketServerFrameHandlerFactory frameHandlerFactory;
     private final Executor executor;
+    private final FrameHandler.ConfigurationCustomizer customizer = new FrameHandler.ConfigurationCustomizer();
     private long asyncSendTimeout = -1;
     private List<Class<?>> deferredEndpointClasses;
     private List<ServerEndpointConfig> deferredEndpointConfigs;
 
     /**
      * Main entry point for {@link JavaxWebSocketServerContainerInitializer}.
-     *
-     * @param webSocketCreatorMapping the {@link WebSocketCreatorMapping} that this container belongs to
+     * @param webSocketMapping the {@link WebSocketMapping} that this container belongs to
      * @param httpClient              the {@link HttpClient} instance to use
      */
-    public JavaxWebSocketServerContainer(WebSocketCreatorMapping webSocketCreatorMapping, HttpClient httpClient, Executor executor)
+    public JavaxWebSocketServerContainer(WebSocketMapping webSocketMapping, HttpClient httpClient, Executor executor)
     {
-        super(new WebSocketCoreClient(httpClient));
-        this._webSocketCreatorMapping = webSocketCreatorMapping;
+        super(() ->
+        {
+            WebSocketCoreClient client = new WebSocketCoreClient(httpClient);
+            if (executor != null && httpClient == null)
+                client.getHttpClient().setExecutor(executor);
+            return client;
+        });
+        this._webSocketMapping = webSocketMapping;
         this.executor = executor;
         this.frameHandlerFactory = new JavaxWebSocketServerFrameHandlerFactory(this);
     }
 
+    @Override
+    public void lifeCycleStopping(LifeCycle context)
+    {
+        ContextHandler contextHandler = (ContextHandler) context;
```

Comment:
If you stop / start a WebApp with this, will the SCI's run again?
If not, then this code is a problem.